SCClassLibrary/Common/Control/Server.sc

```diff
@@ -672,8 +726,30 @@ Server {
 		^Buffer.cachedBufferAt(this, bufnum)
 	}
 
-	defaultGroup {
-		^Group.basicNew(this, 1)
+	// defaultGroups for all clients on this server:
+
+	allClientIDs { ^(0..options.maxLogins-1) }
+
+	// keep defaultGroups for all clients on this server:
+	makeDefaultGroups {
+		defaultGroups = this.allClientIDs.collect { |clientID|
+			Group.basicNew(this, nodeAllocator.numIDs * clientID + 1);
+		};
+		defaultGroup = defaultGroups[clientID];
+	}
+
+	defaultGroupID { ^defaultGroup.nodeID }
+
+	sendDefaultGroups {
+		defaultGroups.do { |defGrp|
+			this.sendMsg("/g_new", defGrp.nodeID.postln);
+		};
+	}
+
+	sendDefaultGroupsForIDs { |ids|
```

Comment:
there are multiple kinds of IDs in this code, so let's be explicit -- recommend rename to `sendDefaultGroupsForClientIDs` and renaming argument to `clientIDs`

Reply:
ok, done